pkg/provision/ec2.go

```diff
@@ -0,0 +1,201 @@
+package provision
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/google/uuid"
+)
+
+// EC2Provisioner contains the EC2 client
+type EC2Provisioner struct {
+	ec2Provisioner *ec2.EC2
+}
+
+// NewEC2Provioner creates an EC2Provisioner and initialises an EC2 client
+func NewEC2Provioner(region string) (*EC2Provisioner, error) {
```

Comment:
I don't like the global state with the env-vars for the AWS keys, since we take them through via flags. Can you look into the session/ec2 API to see if you can pass via argument?

https://github.com/aws/aws-sdk-go/blob/v1.26.8/aws/credentials/static_provider.go#L23

Reply:
Sure, I will do that instead